import/webview/plugin.cpp

```diff
@@ -67,13 +67,40 @@ void SailfishOSWebViewPlugin::initializeEngine(QQmlEngine *engine, const char *u
 
 RawWebView::RawWebView(QQuickItem *parent)
     : QuickMozView(parent)
+    , m_vkbMargin(0.0)
 {
 }
 
 RawWebView::~RawWebView()
 {
 }
 
+qreal RawWebView::virtualKeyboardMargin() const
+{
+    return m_vkbMargin;
+}
+
+void RawWebView::setVirtualKeyboardMargin(qreal vkbMargin)
+{
+    if (m_vkbMargin != vkbMargin) {
+        qreal delta = vkbMargin - m_vkbMargin;
+        m_vkbMargin = vkbMargin;
+        QMargins m = margins();
+        m.setBottom(m.bottom() + delta);
+        setMargins(m);
+        emit virtualKeyboardMarginChanged();
+
+        QVariantMap map;
+        map.insert("imOpen", m_vkbMargin > 0);
+        map.insert("pixelRatio", SailfishOS::WebEngineSettings::instance()->pixelRatio());
+        map.insert("bottomMargin", m.bottom());
+        map.insert("screenWidth", width());
+        map.insert("screenHeight", height());
```

Comment:
This doesn't work.  I was hoping that something like "vkbMargin + height()" could be used to get the "original/non-vkb-open" height, but unfortunately it gives all sorts of bogus results when opening the VKB with different input method hints.